stacker/blueprints/base.py

```diff
@@ -220,7 +222,48 @@ def resolve_variable(var_name, var_def, provided_variable, blueprint_name):
     return value
 
 
+def parse_user_data(variables, raw_user_data, blueprint_name):
+    """Translate a userdata file to into the file contents.
+
+        It supports referencing template variables to create userdata
+        that's supplemented with information from the data, as commonly
+        required when creating EC2 userdata files. Automatically, encodes
+        the data file to base64 after it is processed.
+
+        Args:
+            raw_user_data (str): the user data with the cloud-init info
+
+        Returns:
+            str: The parsed user data, with all the variables values and
+                 refs replaced with their resolved values.
+
+        Raises:
+            MissingVariable: Raised when a variable is in the user_data that
+                             is not given in the blueprint
+
+        """
```

Comment:
This appears to be over indented - I think the whole docstring block is actually.